extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/scan/ScanQueryFrameProcessor.java

```diff
@@ -172,13 +174,26 @@ public static Sequence<Object[]> mappingFunction(Sequence<ScanResultValue> input
     }).map(List::toArray);
   }
 
+  private static ScanQuery prepareScanQuery(@NotNull ScanQuery scanQuery)
+  {
+    if (ScanQuery.Order.NONE.equals(scanQuery.getTimeOrder())) {
```

Comment:
I think we can optimize this further - Consider if we have a scan query without any ordering but with a limit. In that case, we don't need to set the limit to 0 to fetch the results. We can keep the limit as is.
Therefore instead of checking the timeOrder, we'd also need to check the ordering inside, and if its empty or not.

Reply:
Good catch! I've updated the condition